### ISSUE TITLE
Snow-covered blocks are rendered incorrectly with Xali's Enhanced Vanilla (Continuity)

### ISSUE THREAD

**Issue by Lolothepro** [jellysquid3/Sodium#2810]
### Bug Description

Snow-covered blocks rendered incorrectly with Xali's Enhanced Vanilla

With Sodium:
![2024-10-11_20 05 13](https://github.com/user-attachments/assets/fa5d31df-f872-44b1-943f-dd9b0d5881fd)

![2024-10-11_20 14 50](https://github.com/user-attachments/assets/07176cc2-39c9-4bef-acad-b065ce36e8ec)

Without Sodium:
![2024-10-11_20 06 40](https://github.com/user-attachments/assets/ae036c92-1c45-4645-a91f-525067f87d64)

![2024-10-11_20 12 46](https://github.com/user-attachments/assets/4720b647-0ac9-4f83-9c93-cc9a6c8d3ab3)

### Reproduction Steps

Install Xali's Enhanced Vanilla with Continuity
Go in a cold biome and place a grass block

### Log File

https://mclo.gs/xycZea3

### Crash Report

[crash-2024-10-11_20.07.20-client.txt](https://github.com/user-attachments/files/17346103/crash-2024-10-11_20.07.20-client.txt)

**Comment by rilakuma123** [comment]
Originally reported here: https://discord.com/channels/602796788608401408/752637878462578778/1294355955882000405

**Comment by Fyoncle** [comment]
Fixed with Sodium 0.5.12 (Beta 2) backport, can be closed now

**Comment by douira** [comment]
No, because this issue is regarding Sodium 0.6, which is what introduced this issue as part of a performance optimization. Sodium 0.5 doesn't and never has had this issue.

**Comment by Fyoncle** [comment]
> No, because this issue is regarding Sodium 0.6, which is what introduced this issue as part of a performance optimization. Sodium 0.5 doesn't and never has had this issue.

Huh??

**Comment by jellysquid3** [comment]
Closing in favor of #2868 as that issue is more detailed and has a minimal reproduction case.
